### PR TITLE
ADD: js2r-toggle-arrow-function-and-expression(key t a) and some helper functions

### PR DESCRIPTION
**Pull request by futurist**
Also add test for this.

Also fix some potential bugs in file 'js2r-functions.el'
e.g. hard coded "function" looking-at, which will failed with arrow function, instead, using AST.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 elpa
+.tramp_history
```
Comment (NicolasPetton) [inline .gitignore]: This is not required, you could either ignore it in your .git/config/exclude or globally.

File: js2r-functions.el [hunk 2]
```diff
@@ -378,34 +378,105 @@
...
+  (interactive)
+  (save-excursion
+    (ignore-errors
+      (js2r--find-closest-function)
+      (and (looking-at ".+?=>\\s-*{")
+           (not (js2r--point-inside-string-p))))))
+
+(defun js2r--transform-arrow-function-to-expression ()
+  (when (js2r--arrow-function-p)
+    (let ((has-par nil))
```
Comment (NicolasPetton) [inline js2r-functions.el]: rename it to `has-parenthesis` for clarity?

File: js2r-functions.el [hunk 3]
```diff
@@ -378,34 +378,105 @@
...
+        (unless has-par
+          (backward-char 1)
+          (insert ")"))))))
+
+(defun js2r--transform-function-expression-to-arrow ()
+  (when (not (js2r--arrow-function-p))
+    (save-excursion
+      (js2r--find-closest-function)
+      (let* ((pos (point))
+             (fn (js2-node-at-point))
```
Comment (NicolasPetton) [inline js2r-functions.el]: isn't `fn` misleading here?

File: js2r-functions.el [hunk 4]
```diff
@@ -378,34 +378,105 @@
...
+             (params (js2-function-node-params fn))
+             par-start par-end)
```
Comment (NicolasPetton) [inline js2r-functions.el]: Maybe put them in separate lines?

File: js2r-functions.el [hunk 5]
```diff
@@ -378,34 +378,105 @@
...
+        (if (js2r--looking-at-function-declaration)
```
Comment (NicolasPetton) [inline js2r-functions.el]: You could be using `when` instead of `if` here.

**Comment by NicolasPetton** [comment]
It would be nice if you could squash some commits together to get a nicer git history.

**Comment by futurist** [comment]
Thank you for point out! 
Done some cleanups for the code and history.
Oh~, Sorry for outdating your comments :smile:

**Comment by NicolasPetton** [comment]
@futurist can you squash your last commit?

**Comment by futurist** [comment]
ok, done

**Comment by futurist** [comment]
rebased, please review.

**Comment by NicolasPetton** [comment]
Thanks, merged!

**Pull request merged**